iso15118/shared/comm_session.py

```diff
@@ -216,9 +216,17 @@ async def process_message(self, message: bytes):
                 exc.message,
                 self.get_exi_ns(v2gtp_msg.payload_type),
             )
+            logger.error(
+                f"EXI message (ns={self.get_exi_ns(v2gtp_msg.payload_type)}) "
+                f"where validation failed: {v2gtp_msg.payload.hex()}"
+            )
             return
         except EXIDecodingError as exc:
             logger.exception(f"{exc}")
+            logger.error(
+                f"EXI message (ns={self.get_exi_ns(v2gtp_msg.payload_type)}) "
+                f"where error occured: {v2gtp_msg.payload.hex()}"
+            )
```

Comment:
We have an env to enable the logging of the EXI messages:
https://github.com/SwitchEV/iso15118/blob/master/iso15118/shared/settings.py#L18

wouldnt that satisfy your needs?

Reply:
Those messages are just logged in case of an exception. This way it is possible to reproduce the situation in simulation.